python/ml4ir/model/losses/loss_base.py

```diff
@@ -29,6 +29,18 @@ def _loss_fn(y_true, y_pred):
 
         return _loss_fn
 
+    def get_final_activation_op(self):
+        """
+        Returns the final activation layer
+        """
+        return self._final_activation_op()
+
+    def _final_activation_op(self):
+        """
+        Define the final activation layer
+        """
+        raise NotImplementedError
```

Comment:
shouldn't we default to identity activation, and always apply it?  Same effect, right?

Why are there two methods here?  I don't understand why there's `get_final_activation_op` and `_final_activation_op`.  What does it get us?

Reply:
I can default it to identity activation, yes.

You're right. I can merge the two methods.